src/SFA.DAS.Reservations.Functions.LegalEntities.AcceptanceTests/TestServiceProvider.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Memory;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Moq;
+using SFA.DAS.EAS.Account.Api.Client;
+using SFA.DAS.Encoding;
+using SFA.DAS.Providers.Api.Client;
+using SFA.DAS.Reservations.Domain.Reservations;
+
+namespace SFA.DAS.Reservations.Functions.LegalEntities.AcceptanceTests
+{
+    public class TestServiceProvider : IServiceProvider
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public TestServiceProvider()
+        {
+            var serviceCollection = new ServiceCollection();
+            var configuration = GenerateConfiguration();
+
+            var serviceProviderBuilder = new ServiceProviderBuilder(new LoggerFactory(), configuration)
+            {
+                ServiceCollection = serviceCollection
+            };
+            
+            var encodingService = new Mock<IEncodingService>();
+            encodingService.Setup(x => x.Decode(It.Is<string>(s => s.Equals(TestDataValues.NonLevyHashedAccountId)),It.IsAny<EncodingType>())).Returns(TestDataValues.NonLevyAccountId);
```

Comment:
With these setups i'd make sure that its using the correct EncodingType and not an `It.IsAny<>`